### PR TITLE
Improve message retry strategies

### PR DESCRIPTION
**Pull request by joshblum**
patch does the following:

- marks a single conversation after too many attempts instead of the entire outbox
- increases the number of attempts to time out after 15 minutes
- badge inbox items to highlight failed outbox items and add a `LocalMtime` to mark the failure.

in the future we may differentiate 1-1 convs from team convs, indefinitely retrying messages in 1-1 convs.

### CODE REVIEW

File: go/chat/globals/globals.go [hunk 1]
```diff
@@ -35,6 +36,7 @@ type ChatContext struct {
 	LiveLocationTracker  types.LiveLocationTracker        // track live location messages for updates
 	BotCommandManager    types.BotCommandManager          // manages commands from bots in convs
 	UIInboxLoader        types.UIInboxLoader              // manages loading inbox for UI
+	Badger               *badges.Badger                   // app badging
```
Comment (mmaxim) [inline go/chat/globals/globals.go]: How come you did this?

Comment (joshblum) [reply go/chat/globals/globals.go]: needed access in sender.go and server.go, thought it was easier to hang off of g

**Comment by joshblum** [comment]
(ci flake unrelated)

**Comment by mmaxim** [comment]
cc @mlsteele

File: go/chat/inboxsource.go [hunk 2]
```diff
@@ -720,41 +721,117 @@ func (s *HybridInboxSource) markAsReadDeliverLoop(uid gregor1.UID, stopCh chan s
...
+	localUpdates := make(map[string]chat1.LocalConversationUpdate)
+	for _, obr := range obrs {
+		state, err := obr.State.State()
+		if err != nil {
+			s.Debug(ctx, "ApplyLocalChatState: unknown state item: skipping: err: %v", err)
+			continue
+		}
+		if state == chat1.OutboxStateType_ERROR {
+			switch obr.State.Error().Typ {
+			case chat1.OutboxErrorType_MISC,
```
Comment (joshblum) [inline go/chat/inboxsource.go]: comment to avdl

File: go/chat/storage/inbox.go [hunk 3]
```diff
@@ -1892,6 +1895,39 @@ func (i *Inbox) ConversationsUpdate(ctx context.Context, uid gregor1.UID, vers c
 	return i.writeDiskInbox(ctx, uid, ibox)
 }
 
+func (i *Inbox) LocalConversationUpdates(ctx context.Context, uid gregor1.UID,
```
Comment (joshblum) [inline go/chat/storage/inbox.go]: UpdateLocalMtime

File: go/chat/storage/inbox.go [hunk 4]
```diff
@@ -1136,6 +1136,7 @@ func (i *Inbox) NewMessage(ctx context.Context, uid gregor1.UID, vers chat1.Inbo
 	}
 	conv.Conv.ReaderInfo.MaxMsgid = msg.GetMessageID()
 	conv.Conv.ReaderInfo.Mtime = gregor1.ToTime(time.Now())
+	conv.LocalMtime = nil
```
Comment (joshblum) [inline go/chat/storage/inbox.go]: rm

File: go/chat/types/types.go [hunk 5]
```diff
@@ -92,9 +92,13 @@ type RemoteConversation struct {
 	LocalMetadata  *RemoteConversationMetadata `codec:"l"`
 	LocalReadMsgID chat1.MessageID             `codec:"r"`
 	LocalDraft     *string                     `codec:"d"`
+	LocalMtime     *gregor1.Time               `codec:"t"`
 }
 
 func (rc RemoteConversation) GetMtime() gregor1.Time {
+	if rc.LocalMtime != nil {
```
Comment (joshblum) [inline go/chat/types/types.go]: take max instead

**Comment by joshblum** [comment]
merging over unrelated flake

**Pull request merged**